### ISSUE TITLE
63 node limit when using EKS-D from Canonical

### ISSUE THREAD

**Issue by ahachete** [aws/eks-distro#193]
**What happened**:

I recently wrote a blog post titled "[63-Node EKS Cluster running on a Single Instance with Firecracker](https://ongres.com/blog/63-node-eks-cluster-running-on-a-single-instance-with-firecracker/)". The goal of this blog post is to create an EKS cluster on multiple nodes, where each of these nodes is in turn a VM powered by Firecracker. Although possibly insubstantial here, all VMs were running on a single physical host.

As the post mentions, my initial attempt was to create a 100 clusters node. Apparently, I was able to do so. But when listing the nodes (i.e. `kubectl get nodes`) only `63` would be listed. Given that `63=2^6-1`, it might be some kind of "magic number".

The version of EKS-D used is Canonical's ([EKS Distro snap](https://snapcraft.io/eks)). However this might be a particular issue with this distribution or an upstream issue, I'm filing the issue here for completeness.

**What you expected to happen**:

That all 100 nodes would be listed.

**How to reproduce it (as minimally and precisely as possible)**:

All the [source code](https://gitlab.com/ongresinc/blog-posts-src/-/tree/master/202012-firecracker_cloud_image_automation) that lead to the blog post is public and open source. It is fully scripted, and should be easy to reproduce. Review the blog post itself and the first part (linked from the blog post) for further information.

Basically, all 100 VMs are started via the provided scripts. And then Ansible is used to install EKS-D via snap package and configure the nodes. Two different (top-level) roles are used: one for the master, one for the workers. Each worker is pre-assigned a token, used to join the cluster. Please note that no errors were found on the `eks join` commands (99 of such commands where successful) yet only 63 nodes were listed.

**Anything else we need to know?**:

**Environment**:
- EKS Distro Release Channel: EKS-D by Canonical (snap)
- EKS Distro Release: latest

**Comment by abhinavmpandey08** [comment]
Thanks for opening this issue. I will look into it and get back to you.

**Comment by jaxesn** [comment]
@ktsakalozos I think the `eks join` command being referred to here is the snap provided cli.  Any thoughts as to if there is some limit of 63 nodes being applied somewhere on the snap side?

**Comment by ahachete** [comment]
>  I think the eks join command being referred to here is the snap provided cli.

Yes, exactly. You can see how it was called here: https://gitlab.com/ongresinc/blog-posts-src/-/blob/master/202012-firecracker_cloud_image_automation/ansible/roles/eks_node/tasks/01-join_master.yaml

**Comment by ktsakalozos** [comment]
@jaxesn I am not aware of any constraint on the number of nodes we can have in a cluster. However, this is not a something I am testing regularly. I need to run some tests to verify this.
